### PR TITLE
[KOGITO-853] [KOGITO-904] Refactoring Kogito Services

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.redhat.com/browse/KOGITO-853
https://issues.redhat.com/browse/KOGITO-904 (Was a simple change to an annoying bug)

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: deploy/examples/travelapp-kogito-travel-agency.yaml [hunk 1]
```diff
@@ -14,8 +14,8 @@ spec:
...
   build:
     gitSource:
-      contextDir: 05-kogito-travel-agency
-      uri: https://github.com/mswiderski/kogito-travel-agency-tutorial/
+      contextDir: 06-kogito-travel-agency
+      uri: "https://github.com/kiegroup/kogito-travel-agency-tutorial/"
     imageVersion: 0.7.0
     # set your maven nexus repository
-    #mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
+    mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
```
Comment (radtriste) [inline deploy/examples/travelapp-kogito-travel-agency.yaml]: This should stay commented

File: deploy/examples/travelapp-kogito-visas.yaml [hunk 2]
```diff
@@ -14,8 +14,8 @@ spec:
...
   build:
     gitSource:
-      contextDir: 05-kogito-visas
-      uri: https://github.com/mswiderski/kogito-travel-agency-tutorial/
+      contextDir: 06-kogito-visas
+      uri: "https://github.com/kiegroup/kogito-travel-agency-tutorial/"
     imageVersion: 0.7.0
     # set your maven nexus repository
-    #mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
+    mavenMirrorURL: http://nexus3-nexus.apps-crc.testing/repository/maven-public/
```
Comment (radtriste) [inline deploy/examples/travelapp-kogito-visas.yaml]: also there, should be commented

File: deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -16,7 +16,7 @@ metadata:
                 "contextDir": "jbpm-quarkus-example",
                 "uri": "https://github.com/kiegroup/kogito-examples"
               },
-              "imageVersion": "0.8.0"
+              "imageVersion": "0.8.0-rc1"
```
Comment (radtriste) [inline deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml]: shouldn't it be 0.8.0 ?

File: deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -69,7 +69,7 @@ metadata:
     capabilities: Basic Install
     categories: Integration & Delivery
     certified: "false"
-    containerImage: quay.io/kiegroup/kogito-cloud-operator:0.8.0
+    containerImage: quay.io/kiegroup/kogito-cloud-operator:0.8.0-rc1
```
Comment (radtriste) [inline deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml]: also here 0.8.0 ?

File: pkg/infrastructure/services/kafka.go [hunk 5]
```diff
@@ -37,48 +34,46 @@ const (
...
 
-func setKafkaVariables(externalURI string, container *corev1.Container) {
-	if len(externalURI) > 0 {
-		for _, envKey := range managedKafkaKeys {
-			framework.SetEnvVar(envKey, externalURI, container)
-		}
-	}
-}
+// KafkaTopicMessagingType ...
+type KafkaTopicMessagingType string
```
Comment (radtriste) [inline pkg/infrastructure/services/kafka.go]: better to define it into deployer.go or ?

Comment (ricardozanini) [reply pkg/infrastructure/services/kafka.go]: Totally!

File: deploy/crds/app.kiegroup.org_kogitodataindices_crd.yaml [hunk 6]
```diff
@@ -123,159 +222,99 @@ spec:
...
-              description: MemoryRequest is the request of Memory for the container
-              type: string
             replicas:
-              description: Replicas is the number of pod replicas that the Data Index
-                Service will create
+              description: 'Number of replicas that the service will have deployed
+                in the cluster Default value: 1'
               format: int32
-              maximum: 100
+              maximum: 1
```
Comment (sutaakar) [inline deploy/crds/app.kiegroup.org_kogitodataindices_crd.yaml]: Does it mean that Data index shouldn't scale?

Comment (ricardozanini) [reply deploy/crds/app.kiegroup.org_kogitodataindices_crd.yaml]: @cristianonicolai could you chime in? I don't remember honestly if we will have Jobs Service or Data Index as a single pod in the cluster.

File: cmd/kogito/command/completion/completion.go [hunk 7]
```diff
@@ -67,32 +67,32 @@ Zsh:
...
-    shell := args[0]
+	shell := args[0]
 
-    if shell == "bash" {
-        cmd.Root().GenBashCompletion(os.Stdout);
-    } else if shell == "zsh" {
-        cmd.Root().GenZshCompletion(os.Stdout);
-    }
+	if shell == "bash" {
+		cmd.Root().GenBashCompletion(os.Stdout)
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply cmd/kogito/command/completion/completion.go]: yes, I guess David was using VSCode or vi to edit those. My IDE "fixed".

File: deploy/crds/app.kiegroup.org_v1alpha1_kogitodataindex_cr.yaml [hunk 8]
```diff
@@ -8,13 +8,9 @@ spec:
   # number of pods to be deployed
   replicas: 1
   # image to use for this deploy
-  image: "quay.io/kiegroup/kogito-data-index:latest"
+  image:
+    tag: "0.7.0"
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply deploy/crds/app.kiegroup.org_v1alpha1_kogitodataindex_cr.yaml]: we don't have this image yet. When I prepare the PR for the Hub, I'll fix those, since we would have the images for the scorecard testing.

File: deploy/crds/app.kiegroup.org_kogitojobsservices_crd.yaml [hunk 9]
```diff
@@ -136,9 +136,9 @@ spec:
               type: array
             image:
               description: 'Image definition for the service. Example: Domain: quay.io,
-                Namespace: kiegroup, Name: kogito-jobs-service, Tag: latest Defaults
-                to quay.io/kiegroup/kogito-jobs-service:latest On OpenShift an ImageStream
-                will be created in the current namespace pointing to the given image.'
+                Namespace: kiegroup, Name: kogito-jobs-service, Tag: latest On OpenShift
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply deploy/crds/app.kiegroup.org_kogitojobsservices_crd.yaml]: Should be a new line, but the SDK doesn't support it. Fix it. Thanks!

File: cmd/kogito/command/shared/install_data_index.go [hunk 10]
```diff
@@ -38,14 +38,14 @@ func installDefaultDataIndex(cli *client.Client, namespace string) error {
 	kogitoDataIndex := v1alpha1.KogitoDataIndex{
 		ObjectMeta: metav1.ObjectMeta{Name: infrastructure.DefaultDataIndexName, Namespace: namespace},
 		Spec: v1alpha1.KogitoDataIndexSpec{
-			Replicas:       defaultDataIndexReplicas,
-			Image:          infrastructure.DefaultDataIndexImage,
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply cmd/kogito/command/shared/install_data_index.go]: Like @radtriste said, it's defined in the Jobs Service controller now, that's responsible to define how the service supposed to be deployed, including its metadata information (in this case the default image name).

File: deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml [hunk 11]
```diff
@@ -27,17 +27,15 @@ metadata:
             "name": "kogito-data-index"
           },
           "spec": {
-            "cpuLimit": "",
-            "cpuRequest": "",
-            "image": "quay.io/kiegroup/kogito-data-index:latest",
+            "image": {
+              "tag": "0.7.0"
```
Comment (sutaakar) [inline deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml]: Shouldn't be here 0.8.0 or latest?

File: deploy/crds/app.kiegroup.org_kogitodataindices_crd.yaml (hunk 6 above)
Comment (cristianonicolai) [reply deploy/crds/app.kiegroup.org_kogitodataindices_crd.yaml]: job service needs to be single pod for now, scaling needs to be horizontal.
as for data index, it should be fine to scale up.

File: pkg/controller/kogitodataindex/kogitodataindex_controller_test.go [hunk 12]
```diff
@@ -567,16 +481,17 @@ func TestReconcileKogitoDataIndex_updateStatus(t *testing.T) {
...
+					!reflect.DeepEqual(tt.args.instance.Status.Conditions[0].Type, tt.want.Conditions[0].Type) ||
 					!reflect.DeepEqual(tt.args.instance.Status.Conditions[0].Message, tt.want.Conditions[0].Message) {
 					t.Errorf("updateStatus() got status = %v, want status %v", tt.args.instance.Status, tt.want)
 				}
 			}
 		})
 	}
 }
+
+*/
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/controller/kogitodataindex/kogitodataindex_controller_test.go]: This test was removed since won't make sense any more with the new APIs changes. Removed.

File: pkg/infrastructure/services/resources.go [hunk 13]
```diff
@@ -0,0 +1,183 @@
...
+		&deployment.Spec.Template.Spec.Containers[0])
+
+	if infinispanAware.GetInfinispanProperties().UseKogitoInfra || len(infinispanAware.GetInfinispanProperties().URI) > 0 {
+		framework.SetEnvVar(enablePersistenceEnvKey, "true", &deployment.Spec.Template.Spec.Containers[0])
+	}
+	return nil
+}
+
+func (s *serviceDeployer) applyKafkaConfigurations(deployment *appsv1.Deployment, instance v1alpha1.KogitoService) error {
+	URI, err := getKafkaServerURI(instance.GetSpec().(v1alpha1.KafkaAware).GetKafkaProperties(), s.definition.Namespace, s.client)
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/resources.go]: We search for the Kafka Service owned by the Kafka CR, owned by the Kogito Operator if the service is relying on the Infra, otherwise we look for the external URI defined by the user in the CR.

File: pkg/infrastructure/services/manage_status.go [hunk 14]
```diff
@@ -44,8 +40,9 @@ func ManageStatus(instance *v1alpha1.KogitoJobsService, cli *client.Client, errC
 
 	updateStatus := false
 	changed := false
-	updateStatus = updateImageStatus(instance) || updateStatus
-	if changed, err = updateDeploymentStatus(instance, cli); err != nil {
+	updateStatus = updateImageStatus(instance, imageName) || updateStatus
+	var readyReplicas int32
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/manage_status.go]: What did you mean by `envs`? This `readyReplicas` var is to provide a feedback for the user that the service is already provisioned.

File: pkg/infrastructure/services/deployment.go [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	portName    = "http"
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/deployment.go]: That's another JIRA.

File: cmd/kogito/command/install/data_index.go [hunk 16]
```diff
@@ -212,20 +211,23 @@ func (i *installDataIndexCommand) Exec(cmd *cobra.Command, args []string) error
...
+			KogitoServiceSpec: v1alpha1.KogitoServiceSpec{
+				Replicas: i.flags.Replicas,
+				Envs:     shared.FromStringArrayToEnvs(i.flags.Env),
+				Image:    framework.ConvertImageTagToImage(i.flags.image),
+				Resources: v1.ResourceRequirements{
+					Limits:   shared.FromStringArrayToResources(i.flags.Limits),
+					Requests: shared.FromStringArrayToResources(i.flags.Requests),
+				},
+			},
 			HTTPPort:       i.flags.httpPort,
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply cmd/kogito/command/install/data_index.go]: I thought about this possibility, but the Jobs Service doesn't have this option, but: (a) I didn't have the time to open the JIRA yet; (b) I wouldn't want to add to much changes on this JIRA.

Could you please open a JIRA to include this variable also in Jobs Service, and in a way that keep it equal to every service, like `HTTP_PORT` or something like that? Data Index has it's own: `KOGITO_DATA_INDEX_HTTP_PORT`. :(

File: deploy/olm-catalog/kogito-operator/0.8.0-rc1/app.kiegroup.org_kogitodataindices_crd.yaml [hunk 17]
```diff
@@ -78,8 +78,8 @@ spec:
                   description: SaslMechanism defined for the authentication. This
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply deploy/olm-catalog/kogito-operator/0.8.0-rc1/app.kiegroup.org_kogitodataindices_crd.yaml]: I'm deleting them.

File: pkg/infrastructure/services/kafka.go [hunk 18]
```diff
@@ -37,48 +34,36 @@ const (
...
+			return uri, nil
 		}
 	}
+
 	return "", fmt.Errorf(kafkaURINotFoundError)
 }
 
 func getKafkaServerReplicas(kafkaProp v1alpha1.KafkaConnectionProperties, namespace string, client *client.Client) (string, int32, error) {
-	if len(kafkaProp.ExternalURI) <= 0 && infrastructure.IsStrimziAvailable(client) {
+	if len(kafkaProp.ExternalURI) <= 0 {
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/kafka.go]: I didn't get it. What you mean?

File: deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml (hunk 11 above)
Comment (ricardozanini) [reply deploy/olm-catalog/kogito-operator/0.8.0/kogito-operator.v0.8.0.clusterserviceversion.yaml]: See my comment below: https://github.com/kiegroup/kogito-cloud-operator/pull/184#discussion_r386450220

File: pkg/controller/kogitodataindex/kogitodataindex_controller_test.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (sutaakar) [reply pkg/controller/kogitodataindex/kogitodataindex_controller_test.go]: +1

**Comment by ricardozanini** [comment]
Guys many thanks for the detailed review. I believe I addressed the most of them. @radtriste could you please share with me the steps you took to find that "conditions" bug? It's easy to solve, but I need to know where and in which scenario. Thanks!!

File: pkg/controller/kogitoinfra/kafka/resource.go [hunk 19]
```diff
@@ -76,7 +76,7 @@ func CreateRequiredResources(kogitoInfra *v1alpha1.KogitoInfra) (resources map[r
 						"offsets.topic.replication.factor":         defaultReplicas,
 						"transaction.state.log.min.isr":            2,
 						"transaction.state.log.replication.factor": defaultReplicas,
-						"auto.create.topics.enable":                false,
+						"auto.create.topics.enable":                true,
```
Comment (xiezhang7) [inline pkg/controller/kogitoinfra/kafka/resource.go]: If this is set to `true`, kafka topics could be created automatically, but not owned by the kogito custom resources.

Comment (ricardozanini) [reply pkg/controller/kogitoinfra/kafka/resource.go]: Strimzi is not creating the Kafka Topics CR. This is just a flag to auto-create them in the Kafka cluster since Kogito Runtimes and Jobs Service don't create the Kafka Topics CR like Data Index does. Also, I guess we could drop this feature of creating the Topics CR by ourselves and just let Kafka to create them. WDYT?

File: cmd/kogito/command/install/data_index.go (hunk 16 above)
[the remark this answers is not in the document]
Comment (ricardozanini) [reply cmd/kogito/command/install/data_index.go]: Created: https://issues.redhat.com/browse/KOGITO-1336

File: pkg/controller/kogitoinfra/kafka/resource.go (hunk 19 above)
Comment (xiezhang7) [reply pkg/controller/kogitoinfra/kafka/resource.go]: It makes sense if the kafka topics are not supposed to be deleted when data index is uninstalled.

**Comment by radtriste** [comment]
@ricardozanini I got the problem in |
- CR: Install Kogito Data Index
- CR: Process instance events are stored in Data Index
(since you rebased with kogito-1212, the names might have changed, you can find them in `test/features/install_dataindex.feature`)

Gonna re-run the tests for the timeouts I got.

**Comment by radtriste** [comment]
forgot to mention, i the test I am using
- services_image_version and build_image_version 0.8.0-rc1
- examples 0.7.1 (as for now, many are failing on master due to new version of Quarkus)

**Comment by ricardozanini** [comment]
@radtriste just run the Data Index test scenario:

```log
DEBUG=false go test ./test -v -timeout "240m" --godog.tags=""  --godog.concurrency=1 --tests.operator-image-name=quay.io/ricardozanini/kogito-cloud-operator --tests.operator-image-tag=0.8.0 --tests.services-image-version=0.8.0-rc1 --tests.maven-mirror-url=http://nexus3-nexus.apps-crc.testing/repository/maven-public/ --tests.build-image-version=0.8.0-rc1 --tests.smoke features/install_dataindex.feature
{"level":"info","T":"2020-03-02T16:55:33.144-0300","logger":"cucumber-tue0","msg":"Scenario Install Kogito Data Index"}
{"level":"info","T":"2020-03-02T16:55:33.145-0300","logger":"cucumber-tue0","msg":"Step Namespace is created"}
{"level":"info","T":"2020-03-02T16:55:33.145-0300","logger":"cucumber-tue0","msg":"Create namespace cucumber-tue0"}
{"level":"info","T":"2020-03-02T16:55:33.166-0300","logger":"cucumber-tue0","msg":"Step Kogito Operator is deployed with Infinispan and Kafka operators"}
{"level":"info","T":"2020-03-02T16:55:33.176-0300","logger":"cucumber-tue0","msg":"Deploy Operator from yaml files in ../deploy/"}
{"level":"info","T":"2020-03-02T16:55:33.317-0300","logger":"cucumber-tue0","msg":"Subscribing to infinispan operator from source community-operators on channel stable"}
{"level":"info","T":"2020-03-02T16:55:33.348-0300","logger":"cucumber-tue0","msg":"Subscribing to strimzi-kafka-operator operator from source community-operators on channel stable"}
{"level":"info","T":"2020-03-02T16:55:33.375-0300","logger":"cucumber-tue0","msg":"Wait 20m0s for infinispan operator running"}
{"level":"info","T":"2020-03-02T16:55:53.402-0300","logger":"cucumber-tue0","msg":"'infinispan operator running' is successful"}
{"level":"info","T":"2020-03-02T16:55:53.402-0300","logger":"cucumber-tue0","msg":"Wait 20m0s for strimzi-kafka-operator operator running"}
{"level":"info","T":"2020-03-02T16:56:13.407-0300","logger":"cucumber-tue0","msg":"'strimzi-kafka-operator operator running' is successful"}
{"level":"info","T":"2020-03-02T16:56:13.407-0300","logger":"cucumber-tue0","msg":"Step \"CLI\" install Kogito Data Index with 1 replicas"}
{"level":"info","T":"2020-03-02T16:56:13.407-0300","logger":"cucumber-tue0","msg":"cli install Kogito Data Index with 1 replicas"}
{"level":"info","T":"2020-03-02T16:56:13.409-0300","logger":"cucumber-tue0","msg":"Execute CLI [install data-index --image quay.io/kiegroup/kogito-data-index:0.8.0-rc1 -p cucumber-tue0]"}
{"level":"info","T":"2020-03-02T16:56:13.499-0300","logger":"cucumber-tue0","msg":"Step Kogito Data Index has 1 pods running within 10 minutes"}
{"level":"info","T":"2020-03-02T16:56:13.500-0300","logger":"cucumber-tue0","msg":"Wait 10m0s for Kogito data index running"}
{"level":"info","T":"2020-03-02T16:59:03.511-0300","logger":"cucumber-tue0","msg":"'Kogito data index running' is successful"}
{"level":"info","T":"2020-03-02T16:59:03.513-0300","logger":"cucumber-tue0","msg":"Step GraphQL request on service \"kogito-data-index\" is successful within 2 minutes with path \"graphql\" and query:"}
{"level":"info","T":"2020-03-02T16:59:03.517-0300","logger":"cucumber-tue0","msg":"Wait 2m0s for Route kogito-data-index available"}
{"level":"info","T":"2020-03-02T16:59:05.562-0300","logger":"cucumber-tue0","msg":"'Route kogito-data-index available' is successful"}
{"level":"info","T":"2020-03-02T16:59:05.566-0300","logger":"cucumber-tue0","msg":"Wait 2m0s for GraphQL query {\n  ProcessInstances{\n    id\n  }\n} on path 'graphql' to be successful"}
{"level":"info","T":"2020-03-02T16:59:08.723-0300","logger":"cucumber-tue0","msg":"GraphQL response = <nil>"}
{"level":"info","T":"2020-03-02T16:59:08.723-0300","logger":"cucumber-tue0","msg":"'GraphQL query {\n  ProcessInstances{\n    id\n  }\n} on path 'graphql' to be successful' is successful"}
{"level":"info","T":"2020-03-02T16:59:08.993-0300","logger":"cucumber-tue0","msg":"Scenario 'Install Kogito Data Index'. Duration = 3m35.849023219s"}
{"level":"info","T":"2020-03-02T16:59:08.995-0300","logger":"cucumber-tue0","msg":"Delete created namespace cucumber-tue0"}
{"level":"info","T":"2020-03-02T16:59:08.995-0300","logger":"cucumber-tue0","msg":"Delete namespace cucumber-tue0"}
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="godogs" tests="1" skipped="0" failures="0" errors="0" time="3m35.864535357s">
  <testsuite name="Kogito Data Index" tests="1" skipped="0" failures="0" errors="0" time="3m35.579676219s">
    <testcase name="Install Kogito Data Index #1" status="passed" time="3m35.578951963s"></testcase>
  </testsuite>
</testsuites>testing: warning: no tests to run
PASS
ok  	github.com/kiegroup/kogito-cloud-operator/test	215.944s
Tests finished with code 0
```

File: pkg/infrastructure/services/kafka.go (hunk 18 above)
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/kafka.go]: This property is set by the user when they don't want to rely on Kogito Operator to have Kafka deployed, e.g. Kafka is deployed in another namespace, cluster, AWS, GCP, any where in the globe.

To sum up, if this var is filled, we assume Kafka is provided by the user.

File: pkg/infrastructure/services/manage_status.go (hunk 14 above)
[the remark this answers is not in the document]
Comment (spolti) [reply pkg/infrastructure/services/manage_status.go]: yeah, but, cant be it assigned on the if statement?

[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/infrastructure/services/manage_status.go]: It can't since the `if statement has its own scope and we will need that var later in the function, that's why it's only declared there. Plus, we need to specify that the var is `int32` because of the next comparisons. I've changed the `updateStatus` var to be declared and assigned in the same line, though. This is a copy and paste from other function, I missed that. Thanks! :)

**Comment by radtriste** [comment]
@ricardozanini there are problems when deploying directly dataindex via CR (CLI works)

**Comment by radtriste** [comment]
should we set in the test the `UseKogitoInfra` to true ?
Should it not be automatic for dataindex if no configuration is given ?

File: test/framework/kogitojobsservice.go [hunk 20]
```diff
@@ -33,7 +33,7 @@ func InstallKogitoJobsService(namespace string, installerType InstallerType, rep
 	GetLogger(namespace).Infof("%s install Kogito Jobs Service with %d replicas and persistence %v", installerType, replicas, persistence)
 	switch installerType {
 	case CLIInstallerType:
-		return cliInstallKogitoJobsService(namespace, replicas, persistence)
+		return cliInstallKogitoJobsService(namespace, persistence)
```
Comment (radtriste) [inline test/framework/kogitojobsservice.go]: why did you remove the `replicas` here ?

Comment (radtriste) [reply test/framework/kogitojobsservice.go]: nevermind, created https://github.com/ricardozanini/kogito-cloud-operator/pull/9 for you :)

Comment (ricardozanini) [reply test/framework/kogitojobsservice.go]: I've removed because Go lint was screaming at me that the argument wasn't being used. :)

Now that you put some usage for it, we can work on it. :)

**Comment by ricardozanini** [comment]
> should we set in the test the `UseKogitoInfra` to true ?
> Should it not be automatic for dataindex if no configuration is given ?

Yes, that's the old behavior, right? I'll review it.

**Comment by ricardozanini** [comment]
@radtriste I ran a couple times this test:

```
<?xml version="1.0" encoding="UTF-8"?>
<testsuites name="godogs" tests="2" skipped="0" failures="0" errors="0" time="6m17.66764817s">
  <testsuite name="Kogito Data Index" tests="2" skipped="0" failures="0" errors="0" time="6m17.542950828s">
    <testcase name="Install Kogito Data Index #1" status="passed" time="3m3.760940257s"></testcase>
    <testcase name="Install Kogito Data Index #2" status="passed" time="3m13.688448592s"></testcase>
  </testsuite>
</testsuites>testing: warning: no tests to run
PASS
ok  	github.com/kiegroup/kogito-cloud-operator/test	377.702s
Tests finished with code 0
```

Got the same results, but I saw once (and never was able to reproduce), that the GraphQL query returns the "Application Not Ready" page (I'm guessing since the response comes with a `<`).

**Comment by radtriste** [comment]
I see the problem as well, and happens also on master.

I think we should handle that as well in tests. The waitFor is useless for now with graphql if first request(s) is/are returning error(s)...
We should just return false and print error on the graphql waitFor loop.
@sutaakar wdyt ?

**Comment by radtriste** [comment]
Created https://issues.redhat.com/browse/KOGITO-1356
we can discuss there.

btw, this bug is not blocking this PR.

**Comment by radtriste** [comment]
tests passed with images 0.8.0-rc2 and examples from master

**Comment by ricardozanini** [comment]
@radtriste can we merge it?
